lib/internal/modules/cjs/loader.js

```diff
@@ -653,9 +653,13 @@ function getDefaultExtensions() {
   let extensions = ObjectKeys(Module._extensions);
   const tsEnabled = getOptionValue('--experimental-strip-types');
   if (tsEnabled) {
+    // remove .ts and .cts from the default extensions
+    // to avoid extensionless require of .ts and .cts files.
+    // it behaves similarly to how .mjs is handled when --experimental-require-module
+    // is enabled.
     extensions = ArrayPrototypeFilter(extensions, (ext) =>
-      ext !== '.ts' || Module._extensions['.ts'] !== loadTS ||
-      ext !== '.cts' || Module._extensions['.ts'] !== loadCTS,
+      (ext !== '.ts' || Module._extensions['.ts'] !== loadTS) &&
+      (ext !== '.cts' || Module._extensions['.ts'] !== loadCTS),
```

Comment:
```suggestion
      (ext !== '.cts' || Module._extensions['.cts'] !== loadCTS),
```

Context:
https://github.com/nodejs/node/blob/2d1b4a8cf7ce875f0c458941bb40ba6fbdccd42f/lib/internal/modules/cjs/loader.js#L434-L438